server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -812,9 +812,6 @@ public void endVerification(String seed) {
     public RepositoryData getRepositoryData() {
         try {
             return getRepositoryData(latestIndexBlobId());
-        } catch (NoSuchFileException ex) {
```

Comment:
There is no good reason to catch here or below. We break out on generation `-1` and return empty data so whenever we fail to find data for gen. `>= 0` it's a problem and we mustn't ignore it.